api/db-cdr/generate-cdr/libproject/devstart.rb

```diff
@@ -175,23 +197,39 @@ def publish_cdr(cmd_name, args)
         end
       end
 
-      if existing_groups.include?(auth_domain_group_email)
-        common.status "#{auth_domain_group_email} already in ACL, skipping..."
-      else
-        common.status "Adding #{auth_domain_group_email} as a READER..."
-        new_entry = { "groupByEmail" => auth_domain_group_email, "role" => "READER"}
-        json["access"].push(new_entry)
+      unless op.opts.exclude_auth_domain_acl
+        if existing_groups.include?(auth_domain_group_email)
+          common.status "#{auth_domain_group_email} already in ACL, skipping..."
+        else
+          common.status "Adding #{auth_domain_group_email} as a READER..."
+          new_entry = { "groupByEmail" => auth_domain_group_email, "role" => "READER"}
+          json["access"].push(new_entry)
+        end
       end
 
       # if the app SA's in too many groups, it won't gain READER transitively.
       # add it directly, to make sure.
       # See discussion at https://pmi-engteam.slack.com/archives/CHRN2R51N/p1609869521078200?thread_ts=1609796171.063800&cid=CHRN2R51N
-      if existing_users.include?(app_sa)
-        common.status "#{app_sa} already in ACL, skipping..."
-      else
-        common.status "Adding #{app_sa} as a READER..."
-        new_entry = { "userByEmail" => app_sa, "role" => "READER"}
-        json["access"].push(new_entry)
+      unless op.opts.exclude_sa_acl
+        if existing_users.include?(app_sa)
+          common.status "#{app_sa} already in ACL, skipping..."
+        else
+          common.status "Adding #{app_sa} as a READER..."
+          new_entry = { "userByEmail" => app_sa, "role" => "READER"}
+          json["access"].push(new_entry)
+        end
+      end
+
+      if op.opts.additional_reader_group
+        new_group = op.opts.additional_reader_group
+
+        if existing_users.include?(new_group)
```

Comment:
should be `existing_groups.include?` instead

Reply:
right, good catch!